### PR TITLE
feat: added possibility to add, change and remove multiviewers on running production

### PR DESCRIPTION
**Pull request by malmen237**
# What does this do?

This PR makes it possible to add, update and remove multiviewers when a production is running.

Also containing minor fixes:
- Buttons for multiview-config and start production are disabled when no pipelines have been selected.
- Commented out code with TODOs has been removed

View when preparing production:
Screenshot 2024-10-14 at 09 45 11

View when running production:
Screenshot 2024-10-14 at 09 45 29

The configuration looks the same when on a running production and not, but when saving the config you get a warning to confirm that you are about to updated a running config:
Screenshot 2024-10-14 at 12 56 48

### CODE REVIEW

File: src/components/modal/Modal.tsx [hunk 1]
```diff
@@ -8,14 +8,23 @@ type BaseModalProps = {
...
 export type ModalProps = BaseModalProps & PropsWithChildren;
 
-export function Modal({ open, children, outsideClick, className }: ModalProps) {
+export function Modal({
+  open,
+  children,
+  outsideClick,
+  className,
+  disableOutsideClick
+}: ModalProps) {
```
Comment (LucasMaupin) [inline src/components/modal/Modal.tsx:22]: Making outsideClick optional is a better solution than adding another prop I think. Then inside the effect you check "if outsideClick"

**Review by LucasMaupin — APPROVED** [review]

File: src/api/manager/workflow.ts [hunk 2]
```diff
@@ -942,3 +945,262 @@ export async function startProduction(
...
+        value: {
+          success: true,
+          steps: [
+            {
+              step: 'create_multiview',
+              success: false
+            }
+          ]
+        },
+        error: 'Unknown error occured'
```
Comment (Saelmala) [inline src/api/manager/workflow.ts:1039]: Is it possible to return a more descriptive error message than unknown?

File: src/api/manager/workflow.ts [hunk 3]
```diff
@@ -942,3 +945,262 @@ export async function startProduction(
...
+        value: {
+          success: true,
+          steps: [
+            {
+              step: 'update_multiview',
+              success: false
+            }
+          ]
+        },
+        error: 'Unknown error occured'
```
Comment (Saelmala) [inline src/api/manager/workflow.ts:1105]: same here

File: src/api/manager/workflow.ts [hunk 4]
```diff
@@ -942,3 +945,262 @@ export async function startProduction(
...
+        value: {
+          success: true,
+          steps: [
+            {
+              step: 'delete_multiview',
+              success: false
+            }
+          ]
+        },
+        error: 'Unknown error occured'
```
Comment (Saelmala) [inline src/api/manager/workflow.ts:1189]: and here

File: src/app/api/manager/multiviewersOnRunningProduction/[id]/route.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  const { production, updates } = await request.json();
+  return putMultiviewersOnRunningProduction(production, updates)
+    .then((result) => {
+      return new NextResponse(JSON.stringify(result));
+    })
+    .catch((error) => {
+      Log().error(error);
+      const errorResponse = {
+        ok: false,
+        error: 'unexpected'
```
Comment (Saelmala) [inline src/app/api/manager/multiviewersOnRunningProduction/[id]/route.ts:25]: possible with a more descriptive error message?

File: src/app/api/manager/multiviewersOnRunningProduction/[id]/route.ts [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  const { production, removals } = await request.json();
+  return deleteMultiviewersOnRunningProduction(production, removals)
+    .then((result) => {
+      return new NextResponse(JSON.stringify(result));
+    })
+    .catch((error) => {
+      Log().error(error);
+      const errorResponse = {
+        ok: false,
+        error: 'unexpected'
```
Comment (Saelmala) [inline src/app/api/manager/multiviewersOnRunningProduction/[id]/route.ts:47]: same here

File: src/app/api/manager/multiviewersOnRunningProduction/route.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  const { production, additions } = await request.json();
+  return postMultiviewersOnRunningProduction(production, additions)
+    .then((result) => {
+      return new NextResponse(JSON.stringify(result));
+    })
+    .catch((error) => {
+      Log().error(error);
+      const errorResponse = {
+        ok: false,
+        error: 'unexpected'
```
Comment (Saelmala) [inline src/app/api/manager/multiviewersOnRunningProduction/route.ts:22]: possible with a more descriptive error message?

**Pull request merged**